src/elements/content-sharing/ContentSharing.js

```diff
@@ -0,0 +1,165 @@
+    const [errorMessage, setErrorMessage] = useState<?Object>(null);
+
+    const api = new API({
+        apiHost,
+        clientName: CLIENT_NAME_CONTENT_SHARING,
+        id: `${itemType}_${itemID}`,
+        token,
+    });
+
+    const getError = useCallback(
+        (error: $AxiosError<any> | ErrorResponseData) => {
+            let errorObject;
+            if (error.status) {
+                errorObject = contentSharingMessages[CONTENT_SHARING_ERRORS[error.status]];
+            } else if (error.response && error.response.status) {
+                errorObject = contentSharingMessages[CONTENT_SHARING_ERRORS[error.response.status]];
+            } else {
+                errorObject = contentSharingMessages.loadingError;
+            }
+
+            setErrorMessage(errorObject);
+        },
+        [setErrorMessage],
+    );
+
+    useEffect(() => {
+        setItem(null);
+        setSharedLink(null);
+        setCurrentUserID(null);
+    }, [token, itemID, itemType]);
+
+    useEffect(() => {
+        const getItemSuccess = (itemData: ContentSharingItemAPIResponse) => {
+            const { item: itemFromAPI, sharedLink: sharedLinkFromAPI } = convertItemResponse(itemData);
+            setItem(itemFromAPI);
+            setSharedLink(sharedLinkFromAPI);
+            setErrorMessage(null);
+        };
+
+        const getItem = () => {
+            const fields = {
+                fields: [
+                    FIELD_ALLOWED_INVITEE_ROLES,
+                    FIELD_DESCRIPTION,
+                    FIELD_EXTENSION,
+                    FIELD_ID,
+                    FIELD_NAME,
+                    FIELD_PERMISSIONS,
+                    FIELD_SHARED_LINK,
+                    FIELD_SHARED_LINK_FEATURES,
+                    FIELD_ITEM_TYPE,
+                ],
+            };
+            if (itemType === TYPE_FILE) {
+                api.getFileAPI().getFile(itemID, getItemSuccess, getError, fields);
+            } else if (itemType === TYPE_FOLDER) {
+                api.getFolderAPI().getFolderFields(itemID, getItemSuccess, getError, fields);
+            }
+        };
+
+        if (!item && !sharedLink) {
+            getItem();
+        }
+    }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
+
+    useEffect(() => {
+        const getUserSuccess = userData => {
+            const { id, userEnterpriseData } = convertUserResponse(userData);
+            setSharedLink({ ...sharedLink, ...userEnterpriseData });
+            setCurrentUserID(id);
+            setErrorMessage(null);
+        };
+
+        const getUserData = () => {
+            api.getUsersAPI(false).getUser(itemID, getUserSuccess, getError, {
+                fields: [FIELD_ENTERPRISE, FIELD_HOSTNAME],
+            });
+        };
+        if (item && sharedLink && !currentUserID) {
+            getUserData();
+        }
+    }, [api, getError, item, itemID, itemType, sharedLink, currentUserID]);
+
+    const renderElement = () => {
```

Comment:
Does this need to be a separate variable that's declared per-render? It seems like the logic within can be inlined at the bottom of the component.

Reply:
Yes, I believe so. I tried a couple of things here:
- Moving the logic into the `return` statement (worked, but failed linting because of `no-nested-ternary` rule)
- Inlining the logic with the rest of the component (didn't work; the component returned `null` and didn't re-render after making API calls)